src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionParamsWithOptions.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Serialization;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    /// <summary>
+    /// A parameter object that indicates whether the code actions are allowed to offer generating code in
+    /// hidden regions.
+    /// </summary>
+    /// <remarks>
+    /// Code actions normal don't generate code into hidden regions (eg things protected by <c>#line hidden</c> directives
+    /// but Razor generated files are almost entirely hidden, so the Razor client relaxes this rule at will,
+    /// in order to provide better code fixes and refactorings to users. Not all code fixes necessarily support
+    /// this option.
+    /// </remarks>
+    [DataContract]
+    internal class CodeActionParamsWithOptions : CodeActionParams
+    {
+        [DataMember(Name = "allowGenerateInHiddenCode")]
+        public bool AllowGenerateInHiddenCode { get; set; }
```

Comment:
this should be nullable I think right?  Since only Razor provides this.

Reply:
Doesn't the deserializer take care of this, given null, missing, and false all mean the same thing?

If you think it will add clarity though, I'm happy to do it.

Reply:
I'm not sure - typically in the protocol there are newtonsoft json attributes that specify how to handle missing cases.  I have absolutely no idea what happens with [DataMember] ones though.  If missing defaults to false, I'm fine with that.

Reply:
The base class here as DataMember attributes, not Newtonsoft, so I followed suit. NewtonSoft has `DefaultValueHandling` which you can set to `Ignore`, but thats for when serializing the type, so I don't think its relevant here. If we wanted `null` to mean `true` then I think we'd have to add `[DefaultValue(true)]` here, but since we don't, and this isn't marked as a required property, I think we should be fine. I'll test with C# LSP turned on just to confirm though

Reply:
Still works!
<img width="506" alt="image" src="https://user-images.githubusercontent.com/754264/219525473-1cc2ee58-f5a1-44a2-8de3-6664a91cea98.png">